internal/task/manager/stream.go

```diff
@@ -0,0 +1,67 @@
+package task
+
+import (
+	"context"
+	"io"
+	"log"
+
+	basetask "github.com/mikewurtz/taskman/internal/task"
+)
+
+// mergeCancelContexts merges two contexts and cancels when either is done
+func mergeCancelContexts(a, b context.Context) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		select {
+		case <-a.Done():
+		case <-b.Done():
+		}
+		cancel()
+	}()
+	return ctx, cancel
+}
+
+// StreamTaskOutput streams the output of a task to the provided writer function.
+// The writer function is called with chunks of output data. Returns when the task is complete or an error occurs.
+func (tm *TaskManager) StreamTaskOutput(ctx context.Context, taskID string, writer func([]byte) error) error {
+	taskObj, err := tm.getTaskFromMap(taskID)
+	if err != nil {
+		return err
+	}
+
+	// Merge client and server contexts to cancel when either is done
+	// if the either the client or server context is cancelled, we want to stop the stream
+	mergedCtx, cancel := mergeCancelContexts(ctx, tm.ctx)
+	defer cancel()
+
+	var offset int64
+	for {
+		data, newOffset, err := taskObj.ReadOutput(mergedCtx, offset)
+		if err == io.EOF {
```

Comment:
Suggestion: prefer errors.Is
```suggestion
		if errors.Is(err, io.EOF) {
```